compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -953,53 +952,38 @@ public BLangNode transform(TupleTypeDescriptorNode tupleTypeDescriptorNode) {
     @Override
     public BLangNode transform(ErrorTypeDescriptorNode errorTypeDescriptorNode) {
         BLangErrorType errorType = (BLangErrorType) TreeBuilder.createErrorTypeNode();
-        Optional<ErrorTypeParamsNode> typeParam = errorTypeDescriptorNode.errorTypeParamsNode();
+        Optional<TypeParameterNode> typeParam = errorTypeDescriptorNode.errorTypeParamsNode();
         errorType.pos = getPosition(errorTypeDescriptorNode);
         if (typeParam.isPresent()) {
-            ErrorTypeParamsNode typeNode = typeParam.get();
-            BLangType detail = null;
+            TypeParameterNode typeNode = typeParam.get();
+            BLangType detail;
```

Comment:
Do we need this separate variable now? We can directly set to `errorType.detailType` at L961 and L963, right?

Reply:
Done